libr/core/pseudo.c

```diff
@@ -1,7 +1,148 @@
+					memcpy (left, right, rightlen);
+					memmove (comment - leftlen + rightlen, comment, right - comment);
+					memcpy (right - leftlen + rightlen, copy, leftlen);
+				} else if (leftlen < rightlen) {
+					if (linecount < 1 && (copy = (char*) malloc (rightlen))) {
+						memcpy (copy, right, rightlen);
+						// eprintf ("CopyR: '%.*s'\n\n", rightlen, copy);
+						memcpy (right + rightlen - leftlen, left, leftlen);
+						memmove (comment + rightlen - leftlen, comment, right - comment);
+						memcpy (left + rightlen - leftlen, copy, rightlen);
+					} else if (linecount > 0 && (copy = (char*) malloc (linebegin - left))) {
+						memcpy (copy, left, linebegin - left);
+						memset (right - leftpos, ' ', leftpos);
+						*(right - leftpos - 1) = '\n';
+						memcpy (comment + 3, copy, linebegin - left);
+						memset (left, ' ', leftlen);
+					}
+				} else if (leftlen == rightlen && (copy = (char*) malloc (leftlen)) != NULL) {
+					memcpy (copy, right, leftlen);
+					// eprintf ("CopyE: '%.*s'\n\n", leftlen, copy);
+					memcpy (right, left, leftlen);
+					memcpy (left, copy, leftlen);
+				}
+				free (copy);
+			}
+
+			comment = NULL;
+			// >>
+			right = NULL;
+			// <<
+			left = NULL;
+			leftlen = 0;
+			rightlen = 0;
+			linecount = 0;
+			leftpos = 0;
+			leftcolor = 0;
+			rightcolor = 0;
+			commentcolor = 0;
+			type = TYPE_NONE;
+			linebegin = in + 1;
+			continue;
+		} else if (!comment && *in == ';' && *(in + 1) == ' ') {
+			comment = in - 1;
+			comment[1] = '/';
+			comment[2] = '/';
+			while (!IS_WHITESPACE (*(comment - commentcolor))) commentcolor++;
+			commentcolor--;
+			continue;
+		} else if (!comment && type == TYPE_NONE) {
+			if (IS_STRING (in)) {
+				type = TYPE_STR;
+				left = in;
+				while (!IS_WHITESPACE (*(left - leftcolor))) leftcolor++;
+				leftcolor--;
+				leftpos = left - linebegin;
+			} else if (IS_SYMBOL (in)) {
+				type = TYPE_SYM;
+				left = in;
+				while (!IS_WHITESPACE (*(left - leftcolor))) leftcolor++;
+				leftcolor--;
+				leftpos = left - linebegin;
+			}
+			continue;
+		} else if (type == TYPE_STR) {
+			if (!leftlen && left && IS_WHITESPACE (*in)) {
+				leftlen = in - left;
+				// eprintf ("found string left: '%.*s'\n", leftlen, left);
+			} else if (comment && *in == '"' && *(in - 1) != '\\') {
+				if (!right) {
+					right = in;
+					while (!IS_WHITESPACE (*(right - rightcolor))) rightcolor++;
+					rightcolor--;
+				} else {
+					rightlen = in - right + 1;
+					// eprintf ("found string right: '%.*s'\n", rightlen, right);
+				}
+			}
+			continue;
+		} else if (type == TYPE_SYM) {
+			if (!leftlen && left && IS_WHITESPACE (*in)) {
+				leftlen = in - left + 3;
+				// eprintf ("found symbol left: '%.*s'\n", leftlen, left);
+			} else if (comment && *in == '(' && IS_CHAR (*(in - 1)) && !right) {
+				right = in - 1;
+				while (IS_CHAR (*right)) right--;
+				if (*right == ' ') {
+					right--;
+					while (IS_CHAR (*right)) right--;
```

Comment:
braces... but those negative iterations checking if the char != 0 seems wrong to be, because strings are not starting by 0